# Emotions and CNN Project

Goal: create an CNN model which can predict the emotions and use your own pictures to actually recognize your emotions

In [2]:
# Extracting the zipfile 
from zipfile import ZipFile

# specifity the name of the file
zippet = "fer2013.csv.zip"

# open the zip file in read mode
with ZipFile(zippet, 'r') as zip:
    # list all the contents
    zip.printdir()

    # extract all the files
    print('extraction...')
    zip.extractall()
    print('Done!')

File Name                                             Modified             Size
fer2013.csv                                    2019-10-01 04:15:22    301072766
extraction...
Done!


Who can understand anything of this ? it is RGB images with 48x48 pixels. 

Can i trasfomr the 1st row into an image ? 

In [7]:
import pandas as pd

face = pd.read_csv('fer2013.csv')

face.info();

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35887 entries, 0 to 35886
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   emotion  35887 non-null  int64 
 1   pixels   35887 non-null  object
 2   Usage    35887 non-null  object
dtypes: int64(1), object(2)
memory usage: 841.2+ KB


In [27]:
# 1st row and the pixels image
face.loc[0,'pixels'];
# type(face.loc[0,'pixels'])
image1 = list(map(int, face.loc[0,'pixels'].split(" ")))
len(image1)

2304

as we can see the pixels are a 48x48 image since the len is 2304 = 48x48

In [ ]:
import re
import numpy as np
from PIL import Image
from pathlib import Path



# Open image file, slurp the lot
contents = Path('image.txt').read_text()

# Make a list of anything that looks like numbers using a regex...
# ... taking first as height, second as width and remainder as pixels
h, w, *pixels = re.findall(r'[0-9]+', contents)

# Now make pixels into Numpy array of uint8 and reshape to correct height, width and depth
na = np.array(pixels, dtype=np.uint8).reshape((int(h),int(w),3))

# Now make the Numpy array into a PIL Image and save
Image.fromarray(na).save("result.png")

SyntaxError: invalid syntax (520615298.py, line 3)